### PR TITLE
Implement mod alerts when props have flags or are waiting for approval.

### PR DESCRIPTION
**Pull request by markspolakovs**

### CODE REVIEW

File: app/Http/Controllers/PropositionsController.php [hunk 1]
```diff
@@ -119,8 +120,19 @@ public function index()
     		
     		
     	}
-    	
-    	return view('propositions_new', ['fullName' => $user->firstName() . " " . $user->lastName(), 'user' => $viewUser, 'propositions' => $viewPropositions, 'endingSoonPropositions' => $endingSoonPropositions, 'votedPropositions' => $votedPropositions]);
+
+		
+		$modAlerts = array();
+		if($user->role() >= User::ROLE_MODERATOR) {
```
Comment (pavril) [inline app/Http/Controllers/PropositionsController.php]: Only moderators have the right to approve and manage flagged propositions.

File: app/PropositionFactory.php [hunk 2]
```diff
@@ -67,6 +71,9 @@ public function getFlagCount($id) {
 	public function getFlagTypeCount($id, $type) {
 		return Flags::where('proposition', [$id])->where('type', $type)->count();
 	}
+	public function getGlobalFlagCount() {
+		return Flags::count();
```
Comment (pavril) [inline app/PropositionFactory.php]: Flags are not removed from database once moderator took action (see getFlaggedPropositionsExeptUsers in PropositionsFactory).

File: resources/views/layouts_new/authBase.blade.php [hunk 3]
```diff
@@ -65,21 +65,34 @@
 			<p>{{ Lang::get('messages.notifications.available_in_en') }} <a href="{{ route('profile.language.set', ['en']) }}" class="alert-link">{{ Lang::get('messages.languages.en') }}</a>!</p>
 			@endif
 		</div>
-	</div>
-	
-	
+
+
+		@if (isset($modAlerts["approval"]) AND $modAlerts["approval"])
```
Comment (pavril) [inline resources/views/layouts_new/authBase.blade.php]: Unclear if statement: `$modAlerts["approval"]` must not be empty?

File: resources/views/layouts_new/authBase.blade.php [hunk 4]
```diff
@@ -65,21 +65,34 @@
...
+            <div class="alert alert-warning" role="alert" id="mod-approval">
+                <p><a href="{{ route('moderator.approval') }}" class="alert-link">{{ Lang::get('messages.notifications.moderator_approval_queue')  }}</a></p>
+            </div>
+        @endif
+            @if (isset($modAlerts["flag"]) AND $modAlerts["flag"])
```
Comment (pavril) [inline resources/views/layouts_new/authBase.blade.php]: Unclear if statement: `$modAlerts["flag"]` must not be empty?

File: app/PropositionFactory.php (hunk 2 above)
Comment (markspolakovs) [reply app/PropositionFactory.php]: @pavril So what should this be? `Flags::where('status', '<', 3)->whereNotIn('proposer_id', [$id])->groupBy('proposition')->join('propositions', 'flags.proposition', '=', 'propositions.propositionId')->count()`? Something like that?

Comment (pavril) [reply app/PropositionFactory.php]: Yes! You can remove `groupBy` and `join` as you are only counting flags and don't need any information about propositions

File: resources/views/layouts_new/authBase.blade.php (hunk 3 above)
Comment (markspolakovs) [reply resources/views/layouts_new/authBase.blade.php]: `$modAlerts["approval"]` won't be set if the user isn't a moderator.

Comment (pavril) [reply resources/views/layouts_new/authBase.blade.php]: So the message is shown if only `$modAlerts["approval"]` is set and is not empty?
Maybe consider using `@if (isset($modAlerts["approval"]) == true AND empty($modAlerts["approval"]) == true)`

**Comment by markspolakovs** [comment]
This PR should probably be squashed when you merge it

File: app/PropositionFactory.php (hunk 2 above)
Comment (markspolakovs) [reply app/PropositionFactory.php]: image
Wait a minute. I don't see a `status` column anywhere.
image

File: app/PropositionFactory.php [hunk 5]
```diff
@@ -50,17 +50,16 @@ public function search($term, $pagination) {
...
-		return Proposition::whereStatus(2)->count();
-	}
 	
 	public function getQueuedPropositions() {
 		return Proposition::whereStatus(2)->orderBy('deadline', 'asc')->get();
 	}
 	public function getQueuedPropositionsExeptUsers($id) {
 		return Proposition::whereStatus(2)->whereNotIn('proposer_id', [$id])->orderBy('deadline', 'asc')->get();
 	}
+	public function getQueuedExceptUserCount($id) {
```
Comment (pavril) [inline app/PropositionFactory.php]: Just to have clear functions, can you rename `getQueuedExceptUserCount` to `getQueuedPropositionsExceptUserCount`?

File: app/PropositionFactory.php (hunk 2 above)
Comment (pavril) [reply app/PropositionFactory.php]: Yes, currently fixing the bug, needed to join with propositions model to filter proposition status. My bad ...

**Comment by pavril** [comment]
This branch isn't merged directly into master. However, it is currently merged into master on a commit (72f7685).

**Pull request closed without merging**